### PR TITLE
FOUNDATIONS: Upgraded Tasks to V3

### LINKED ISSUE #66: FOUNDATIONS: checkout@v3 and setup-dotnet@v3 tasks
Upgrade tasks to V3 to get rid of build warnings
```
Node.js 12 actions are deprecated. 
Please update the following actions to use Node.js 16: actions/checkout@v2, actions/setup-dotnet@v1. 
For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.
```

### PR DESCRIPTION
**Pull request by cjdutoit**
Closes #66

### CODE REVIEW

**Comment by BrianLParker** [comment]
LGTM

File: ADotNet.Infrastructure.Build/Program.cs [hunk 1]
```diff
@@ -50,19 +50,18 @@ static void Main(string[] args)
...
+                            new SetupDotNetTaskV3
                             {
                                 Name = "Setup .Net",
 
-                                TargetDotNetVersion = new TargetDotNetVersion
+                                TargetDotNetVersion = new TargetDotNetVersionV3
                                 {
-                                    DotNetVersion = "7.0.201",
-                                    IncludePrerelease = false
+                                    DotNetVersion = "7.0.201"
```
Comment (glhays) [inline ADotNet.Infrastructure.Build/Program.cs:64]: Current version is "7.0.203", is there reasoning to not use 7.0.*

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -31,12 +31,11 @@ jobs:
     runs-on: windows-2019
     steps:
     - name: Check out
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
     - name: Setup .Net
-      uses: actions/setup-dotnet@v1
+      uses: actions/setup-dotnet@v3
       with:
         dotnet-version: 7.0.201
```
Comment (glhays) [inline .github/workflows/dotnet.yml:38]: Same here

File: ADotNet.Infrastructure.Build/Program.cs (hunk 1 above)
Comment (BrianLParker) [reply ADotNet.Infrastructure.Build/Program.cs:64]: I think the correct syntax is 7.0.x
I have used this on my own projects using `ADotNet` and it seems to work, It builds latest non pre release.

Comment (glhays) [reply ADotNet.Infrastructure.Build/Program.cs:64]: Thanks Brian your correct my bad 7.0.* is never going to work 👍 @cjdutoit said he received some errors with the latest version so he reverted back to 7.0.201, I have used the syntax 7.0.x as well.

**Pull request merged**